cpp/src/strings/extract/extract_all.cu

```diff
@@ -49,31 +45,32 @@ namespace {
  * The `d_offsets` are pre-computed to identify the location of where each
  * string's output groups are to be written.
  */
-template <int stack_size>
 struct extract_fn {
   column_device_view const d_strings;
-  reprog_device d_prog;
   offset_type const* d_offsets;
   string_index_pair* d_indices;
 
-  __device__ void operator()(size_type idx)
+  __device__ void operator()(size_type const idx,
+                             reprog_device const d_prog,
+                             int32_t const prog_idx)
```

Comment:
Why is this an `int32_t` rather than a `size_type`? Other parts of this PR have moved towards greater consistency in using `size_type`.

Reply:
On further reading, I guess you are using `int32_t` everywhere now for the number of threads or the number of distinct reprog instances while reserving `size_type` for its usual meaning of the max column size in libcudf?

Reply:
Yes, it is not a `size_type` since the value is not really a size.